### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by omeid** [hashicorp/terraform-provider-aws#28755]
### Terraform Core Version

1.3.7

### AWS Provider Version

4.49.0

### Affected Resource(s)

aws_route_table's route object

### Expected Behavior

Removal of aws_route_table's route object should remove the route.

### Actual Behavior

noop.

### Relevant Error/Panic Output Snippet

```shell
N/A
```

### Terraform Configuration Files

resource "aws_route_table" "test" {
  vpc_id = aws_vpc.test.id

  route { 
    cidr_block = "0.0.0.0/0"
    gateway_id = aws_internet_gateway.test.id
  }

  tags = {
    Name = var.name
  }

  depends_on = [ aws_internet_gateway.test]
}

### Steps to Reproduce

1. Apply the configuration provided.
2. Remove "route" entry.
3. apply the config again. Terraform reports 0 changes.

### Debug Output

N/A.

### Would you like to implement a fix?

None

**Comment by trevorrea** [comment]
Hi,

I can't comment on your issue but it's much easier to use the separate https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route resource rather then defining the routes inline. Defining anything inline is a pain and never works correctly. Using the route resource will work the way you expect.

Regards,
Trevor

**Comment by omeid** [comment]
Thank you Trevor,

I have opted for separate ones.

The issue is when refactoring other's code. It can leave routes behind that can potentially increase attack surface and create a security risk.

**Comment by trevorrea** [comment]
Ah.. well I guess if you want to recover from the situation you're in then you could create new route tables and create the routes using the separate route resources. Then update the route tables associations to point to the new route tables and once that's complete and working then remove the original route tables.

**Comment by omeid** [comment]
Yes, that is a good approach to handle it without reaching for the UI. However, it requires that you know this issue exists and remember it!

**Comment by justinretzolk** [comment]
Hey @omeid 👋 In this case, this is expected behavior, as called out in the [`route` argument description](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route_table#route):

> This argument is processed in [attribute-as-blocks mode](https://www.terraform.io/docs/configuration/attr-as-blocks.html). This means that omitting this argument is interpreted as ignoring any existing routes. To remove all managed routes an empty list should be specified.

If you set the `route` argument to an empty list (`route = []`), the route(s) will be removed.

**Comment by omeid** [comment]
This is a bug. Documenting this bug as expected behaviour does not change it.

Ignoring should be explicit like every other attribute. This is very uprising and unexpected. I would be very surprised if this hasn't already caused security problems for people, think the routes for a given subnet is removed but they could be even public still.

**Comment by omeid** [comment]
Not stable. Still a bug.
